merkle/index.go

```diff
@@ -0,0 +1,78 @@
+package merkle
+
+import "sort"
+
+// siblingIndecies returns indecirs of sibling elements
+func siblingIndecies(idxs []uint64) []uint64 {
+	indicesCount := len(idxs)
+	siblings := make([]uint64, indicesCount)
+	for i := 0; i < indicesCount; i++ {
+		siblings[i] = siblingIndex(idxs[i])
+	}
+	return siblings
+}
+
+// parentIndecies returns indecies of parent elements
+func parentIndecies(idxs []uint64) []uint64 {
+	indicesCount := len(idxs)
+	parents := make([]uint64, indicesCount)
+	for i := 0; i < indicesCount; i++ {
+		parents[i] = parentIndex(idxs[i])
+	}
+	parents = removeDuplicateIndices(parents)
+	return parents
+}
+
+// siblingIndex returns index of a sibling element
+func siblingIndex(idx uint64) uint64 {
+	if isLeftIndex(idx) {
+		// Right sibling index
+		return idx + 1
+	}
+	// Left sibling index
+	return idx - 1
```

Comment:
I think you can replace all this by using XOR
```suggestion
	return idx ^ 1
```

Reply:
fixed in [this](https://github.com/ComposableFi/go-merkle-trees/pull/22/commits/2e360cdbbfc6c3e12c3c206b0c3b3d92e91c702c)